python/src/boombox/boombox.py

```diff
@@ -0,0 +1,471 @@
+        ------
+        AudioPacket or VideoPacket
+            Raw media packets produced by Boombox.
+
+        Raises
+        ------
+        RuntimeError
+            If Boombox's output was not defined by an :py:class:`.Array` endpoint.
+        """
+        while True:
+            match self._call(Atom("produce_packet")):
+                case (Atom("ok"), packet):
+                    yield self._deserialize_packet(packet)
+                case (Atom("finished"), packet):
+                    yield self._deserialize_packet(packet)
+                    break
+                case (Atom("error"), Atom("incompatible_mode")):
+                    raise RuntimeError("Output not defined with an Array endpoint.")
+                case other:
+                    raise RuntimeError(f"Unknown response: {other}")
+
+    def write(self, packet: AudioPacket | VideoPacket) -> bool:
+        """Write packets to Boombox.
+
+        Enabled only if Boombox has been initialized with input defined with an
+        :py:class:`.Array` endpoint.
+
+        This method provides Boombox with a packet to process and returns once
+        Boombox is ready to accept the next packet.
+
+        Parameters
+        ----------
+        packet : AudioPacket or VideoPacket
+            Raw media packet to be consumed by Boombox.
+
+        Returns
+        -------
+        finished : bool
+            Informs if Boombox has finished writing to it's output and won't
+            accept any more packets to consume.
+
+        Raises
+        ------
+        RuntimeError
+            If Boombox's input was not defined with an :py:class:`.Array` endpoint.
+        """
+        serialized_packet = self._serialize_packet(packet)
+
+        match self._call((Atom("consume_packet"), serialized_packet)):
+            case Atom("finished"):
+                return True
+            case Atom("ok"):
+                return False
+            case (Atom("error"), Atom("incompatible_mode")):
+                raise RuntimeError("Input not defined with an Array endpoint.")
+            case other:
+                raise RuntimeError(f"Unknown response: {other}")
+
+    def close(self, wait: bool = False, kill: bool = False) -> None:
+        """Closes Boombox for writing.
+
+        Enabled only if Boombox has been initialized with input defined with an
+        :py:class:`.Array` endpoint.
+
+        This method informs Boombox that it shouldn't expect any more packets.
+
+        Parameters
+        ----------
+        wait : bool, default=False
+            Determines whether this method should wait until Boombox finishes
+            it's operation and only then return, or if it should return
+            immediately and let Boombox finish in the background.
+        kill : bool, default=False
+            Determines whether Boombox should be killed without waiting for it
+            to gracefully finish it's operation. If True this method will
+            return immediately.
```

Comment:
I am not sure it's perfectly clear what happens if we set both `wait` and `kill` to `True`